src/main/java/luke/Deadline.java

```diff
@@ -0,0 +1,41 @@
+package luke;
+
+import java.time.LocalDate;
+
+public class Deadline extends Task {
+    private LocalDate by;
```

Comment:
byDate may be more applicable as a variable name. Given that you are storing it as a LocalDate. Future programmers will know it doesn't store for instance Tuesday or 3 pm.  Applicable to Event as well.